[BUG] Apps program name cannot include "-"

### Description / Steps to reproduce the issue

If we rename hello to hello-world:
<img width="732" height="234" alt="Image" src="https://github.com/user-attachments/assets/595e5d2e-7ac0-4c4c-8998-3c521b95e800" />

the build will fail:

```
In file included from builtin_list.c:34:
builtin_proto.h:6:10: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘-’ token
    6 | int hello-world_main(int argc, char *argv[]);
      |          ^
In file included from builtin_list.c:50:
builtin_list.h:6:29: error: ‘hello’ undeclared here (not in a function)
    6 | { "hello-world", 100, 2048, hello-world_main },
      |                             ^~~~~
builtin_list.h:6:35: error: ‘world_main’ undeclared here (not in a function); did you mean ‘dd_main’?
    6 | { "hello-world", 100, 2048, hello-world_main },
      |                                   ^~~~~~~~~~
      |                                   dd_main
make[2]: *** [/home/alan/nuttxspace/apps/Application.mk:239: builtin_list.c.home.alan.nuttxspace.apps.builtin.o] Error 1
make[2]: *** Waiting for unfinished jobs....
<command-line>: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘-’ token
hello_main.c:38:5: note: in expansion of macro ‘main’
   38 | int main(int argc, FAR char *argv[])
      |     ^~~~
```


### On which OS does this issue occur?

[OS: Linux]

### What is the version of your OS?

Ubuntu 24.04

### NuttX Version

mainline

### Issue Architecture

[Arch: all]

### Issue Area

[Area: Other]

### Host information

_No response_

### Verification

- [x] I have verified before submitting the report.

--- a/cmake/nuttx_add_application.cmake
+++ b/cmake/nuttx_add_application.cmake
@@ -111,6 +111,8 @@ function(nuttx_add_application)
     return()
   endif()
 
+  string(REPLACE "-" "_" NAME_SYM "${NAME}")
+
   # check if SRCS exist
   if(SRCS)
     file(GLOB SRCS_EXIST ${SRCS})
@@ -257,7 +259,7 @@ function(nuttx_add_application)
         set_property(
           SOURCE ${MAIN_SRC}
           APPEND
-          PROPERTY COMPILE_DEFINITIONS main=${NAME}_main)
+          PROPERTY COMPILE_DEFINITIONS main=${NAME_SYM}_main)
       endif()
     endif()
 
@@ -275,7 +277,7 @@ function(nuttx_add_application)
 
   # store parameters into properties (used during builtin list generation)
 
-  set_target_properties(${TARGET} PROPERTIES APP_MAIN ${NAME}_main)
+  set_target_properties(${TARGET} PROPERTIES APP_MAIN ${NAME_SYM}_main)
   set_target_properties(${TARGET} PROPERTIES APP_NAME ${NAME})
 
   if(PRIORITY)

